### PR TITLE
feat: Migrate `identifier-results` x-module to composition API

### PR DESCRIPTION
**Pull request by andreadlgdo**
# Pull request template
 
Describe the purpose of the change, the specific changes done in detail, and the issue you have fixed.

## Motivation and context

- [ ] Dependencies. If any, specify:
- [x] Open issue. If applicable, link: [EMP-4312](https://searchbroker.atlassian.net/browse/EMP-4312)

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that causes existing functionality to not work as expected)
- [ ] Change requires a documentation update

## What is the destination branch of this PR?

- [x] `Main`
- [ ] Other. Specify:

## How has this been tested?

Tests performed according to [testing guidelines](./contributing/tests.md): 

## Checklist:

- [ ] My code follows the **[style guidelines](./CONTRIBUTING.md#style-guides)** of this project.
- [ ] I have performed a **self-review** on my own code.
- [ ] I have **commented** my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the **[documentation](./CONTRIBUTING.md#documentation-style-guide)**.
- [ ] My changes generate **no new warnings**.
- [ ] I have added **tests** that prove my fix is effective or that my feature works.
- [ ] New and existing **unit tests pass locally** with my changes.

[EMP-4312]: https://searchbroker.atlassian.net/browse/EMP-4312?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/x-components/src/x-modules/identifier-results/components/identifier-result.vue [hunk 1]
```diff
@@ -9,64 +9,71 @@
...
+       * Highlights the matching part of the identifier result with the query from the state.
+       *
+       * @returns String - The identifier result s query with the matching part inside a `<span>` tag.
+       * @public
+       */
+      const highlightedQueryHTML = computed((): string => {
+        const identifierValue = props.result.identifier?.value ?? '';
+        if (identifierValue && identifierHighlightRegexp.value) {
+          return identifierValue.replace(
+            identifierHighlightRegexp.value,
```
Comment (lauramargar) [inline packages/x-components/src/x-modules/identifier-results/components/identifier-result.vue:64]: Try to fix this warning

Screenshot 2024-06-17 at 16 02 44

Maybe you should assign `RegExp` type

File: packages/x-components/src/x-modules/identifier-results/components/identifier-result.vue [hunk 2]
```diff
@@ -9,64 +9,71 @@
...
-     */
-    @Getter('identifierResults', 'identifierHighlightRegexp')
-    public identifierHighlightRegexp!: RegExp;
+      /**
+       * Highlights the matching part of the identifier result with the query from the state.
+       *
+       * @returns String - The identifier result s query with the matching part inside a `<span>` tag.
+       * @public
+       */
+      const highlightedQueryHTML = computed((): string => {
```
Comment (lauramargar) [inline packages/x-components/src/x-modules/identifier-results/components/identifier-result.vue:60]: The return type (`:string`) is already inferred. You can remove it.

File: packages/x-components/src/x-modules/identifier-results/components/identifier-result.vue [hunk 3]
```diff
@@ -9,64 +9,71 @@
 
 <script lang="ts">
   import { Result } from '@empathyco/x-types';
-  import Vue from 'vue';
-  import { Component, Prop } from 'vue-property-decorator';
-  import { Getter, State } from '../../../components/decorators/store.decorators';
-  import { xComponentMixin } from '../../../components/x-component.mixin';
+  import { computed, defineComponent, PropType } from 'vue';
   import { identifierResultsXModule } from '../x-module';
+  import { useGetter, useState } from '../../../composables';
```
Comment (joseacabaneros) [inline packages/x-components/src/x-modules/identifier-results/components/identifier-result.vue:14]: Avoid importing from composables barrel

File: packages/x-components/src/x-modules/identifier-results/components/identifier-results.vue [hunk 4]
```diff
@@ -17,13 +17,12 @@
...
-  import { Component, Prop, Provide } from 'vue-property-decorator';
-  import Vue from 'vue';
-  import { State } from '../../../components/decorators/store.decorators';
-  import { xComponentMixin } from '../../../components/x-component.mixin';
+  import { computed, defineComponent, provide } from 'vue';
   import { PropsWithType } from '../../../utils/types';
   import { XEventsTypes } from '../../../wiring/events.types';
   import { identifierResultsXModule } from '../x-module';
+  import { AnimationProp } from '../../../types';
+  import { useState } from '../../../composables';
```
Comment (joseacabaneros) [inline packages/x-components/src/x-modules/identifier-results/components/identifier-results.vue:25]: Avoid barrel

File: packages/x-components/src/x-modules/identifier-results/x-module.ts [hunk 5]
```diff
@@ -1,4 +1,5 @@
 import { XModule } from '../x-modules.types';
+import { XPlugin } from '../../plugins/index';
```
Comment (joseacabaneros) [inline packages/x-components/src/x-modules/identifier-results/x-module.ts:2]: Avoid barrel

File: packages/x-components/src/plugins/__tests__/x-plugin-alias.spec.ts [hunk 6]
```diff
@@ -45,7 +45,7 @@ describe('testing plugin alias', () => {
         search: ''
       },
       status: {
-        identifierResults: undefined,
+        identifierResults: 'initial', // It is already registered by the `relatedTagsXModule` import itself
```
Comment (joseacabaneros) [inline packages/x-components/src/plugins/__tests__/x-plugin-alias.spec.ts:48]: Fix typo `registered by the `identifierResultsXModule` import itself`

File: packages/_vue3-migration-test/src/x-modules/identifier-results/x-module.ts [hunk 7]
```diff
@@ -0,0 +1,40 @@
+import { PrivateXModuleOptions } from '../../../../x-components/src/plugins/x-plugin.types';
+import { IdentifierResultsXModule } from '../../../../x-components/src/x-modules/identifier-results/x-module';
+export const identifierResultsXModule: PrivateXModuleOptions<IdentifierResultsXModule> = {
+  storeModule: {
```
Comment (joseacabaneros) [inline packages/_vue3-migration-test/src/x-modules/identifier-results/x-module.ts:4]:
```TS
  storeModule: {
    state: {
      identifierResults: ['123A', '123B', '123C', '123D'].map(id => createResultStub(id)),
      query: 'test'
    }
  }
```

Would it be enough?

**Review by lauramargar — APPROVED** [review]

**Pull request merged**